src/app/components/PriceChart.tsx

```diff
@@ -316,3 +354,126 @@ export function PriceChart() {
     </>
   );
 }
+
+export function PairInfoTab() {
+  const t = useTranslations();
+  const { pairsList } = useAppSelector((state) => state.rewardSlice);
+  const selectedPairAddress = useAppSelector(
+    (state) => state.pairSelector.address
+  );
+  const pairInfo = pairsList.find(
+    (pairInfo) => pairInfo.address === selectedPairAddress
+  );
+
+  let name, address, orderReceiptAddress, token1, token2;
+  if (pairInfo) {
+    ({ name, address, orderReceiptAddress, token1, token2 } = pairInfo);
+  } else {
+    return "Selected pair not found in pairsList";
+  }
+
+  const shortenAddress = (address: any) => {
+    if (!address) return "";
+    return `${address.slice(0, 40)}...`;
+  };
+
+  return (
```

Comment:
This full component could be split up into 3 sub-components for better readibility and maintainability. For example:
- AddressAndLabel
- CoinInfo
- CopyIcon

I shared more details via discord DM.